flow-preferences/src/main/java/com/fredporciuncula/flow/preferences/FlowSharedPreferences.kt

```diff
@@ -4,70 +4,149 @@ import android.content.SharedPreferences
 import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.flow.Flow
 import kotlin.coroutines.CoroutineContext
+import kotlin.coroutines.EmptyCoroutineContext
 
 internal typealias KeyFlow = Flow<String?>
 
+/**
+ * Factory for [Preference] objects based on an instance of [SharedPreferences]. Commit operations happen on coroutineContext, which is [Dispatchers.IO] by default.
+ */
 class FlowSharedPreferences @JvmOverloads constructor(
   val sharedPreferences: SharedPreferences,
-  val coroutineContext: CoroutineContext = Dispatchers.IO
+  val coroutineContext: CoroutineContext = EmptyCoroutineContext,
 ) {
 
   private val keyFlow: KeyFlow = sharedPreferences.keyFlow
 
+  /**
+   * Creates a [Preference] of type [Int] with the given key and defaultValue.
+   *
+   * Note: By default, all operations will be executed on [Dispatchers.IO]
```

Comment:
I don't think it's necessary to repeat this on every method as this is already mentioned in the class documentation. Also, it doesn't really apply to **all** operations.